Fixed several printf arguments not having the expected type.

From: Roman Donchenko <roman.donchenko@itseez.com>

This patch is originally from <https://kforge.ros.org/openni/drivers/>,
slightly modified by me.
---
 Source/OpenNI/XnProfiling.cpp        |    8 ++++----
 Source/Utils/niLicense/niLicense.cpp |    4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/Source/OpenNI/XnProfiling.cpp b/Source/OpenNI/XnProfiling.cpp
index f9874fa..93dd0a2 100644
--- a/Source/OpenNI/XnProfiling.cpp
+++ b/Source/OpenNI/XnProfiling.cpp
@@ -87,8 +87,8 @@ XN_THREAD_PROC xnProfilingThread(XN_THREAD_PARAM /*pThreadParam*/)
 		// print profiled sections
 		nReportChars = 0;
 		nReportChars += sprintf(csReport + nReportChars, "Profiling Report:\n");
-		nReportChars += sprintf(csReport + nReportChars, "%-*s %-5s %-6s %-9s %-7s\n", g_ProfilingData.nMaxSectionName, "TaskName", "Times", "% Time", "TotalTime", "AvgTime");
-		nReportChars += sprintf(csReport + nReportChars, "%-*s %-5s %-6s %-9s %-7s\n", g_ProfilingData.nMaxSectionName, "========", "=====", "======", "=========", "=======");
+		nReportChars += sprintf(csReport + nReportChars, "%-*s %-5s %-6s %-9s %-7s\n", (int)g_ProfilingData.nMaxSectionName, "TaskName", "Times", "% Time", "TotalTime", "AvgTime");
+		nReportChars += sprintf(csReport + nReportChars, "%-*s %-5s %-6s %-9s %-7s\n", (int)g_ProfilingData.nMaxSectionName, "========", "=====", "======", "=========", "=======");
 
 		XnUInt64 nTotalTime = 0;
 
@@ -104,7 +104,7 @@ XN_THREAD_PROC xnProfilingThread(XN_THREAD_PARAM /*pThreadParam*/)
 				nAvgTime = pSection->nTotalTime / pSection->nTimesExecuted;
 			}
 
-			nReportChars += sprintf(csReport + nReportChars, "%-*s %5u %6.2f %9llu %7llu\n", g_ProfilingData.nMaxSectionName, 
+			nReportChars += sprintf(csReport + nReportChars, "%-*s %5u %6.2f %9llu %7llu\n", (int)g_ProfilingData.nMaxSectionName,
 				pSection->csName, pSection->nTimesExecuted, dCPUPercentage, pSection->nTotalTime, nAvgTime);
 
 			if (pSection->nIndentation == 0)
@@ -118,7 +118,7 @@ XN_THREAD_PROC xnProfilingThread(XN_THREAD_PARAM /*pThreadParam*/)
 		// print total
 		XnDouble dCPUPercentage = ((XnDouble)nTotalTime) / (nNow - nLastTime) * 100.0;
 		nReportChars += sprintf(csReport + nReportChars, "%-*s %5s %6.2f %9llu %7s\n", 
-			g_ProfilingData.nMaxSectionName, "*** Total ***", "-", dCPUPercentage, nTotalTime, "-");
+			(int)g_ProfilingData.nMaxSectionName, "*** Total ***", "-", dCPUPercentage, nTotalTime, "-");
 
 		xnLogVerbose(XN_MASK_PROFILING, "%s", csReport);
 
diff --git a/Source/Utils/niLicense/niLicense.cpp b/Source/Utils/niLicense/niLicense.cpp
index b16659c..09b15da 100644
--- a/Source/Utils/niLicense/niLicense.cpp
+++ b/Source/Utils/niLicense/niLicense.cpp
@@ -119,14 +119,14 @@ int main(int argc, char* argv[])
 		nRetVal = xnOSStrNCopy(license.strVendor, strVendor, strlen(strVendor) + 1, sizeof(license.strVendor));
 		if (nRetVal != XN_STATUS_OK)
 		{
-			printf("Error: Vendor name is too long (should be up to %u characters).", sizeof(license.strVendor) - 1);
+			printf("Error: Vendor name is too long (should be up to %lu characters).", (unsigned long) (sizeof(license.strVendor) - 1));
 			return -1;
 		}
 
 		nRetVal = xnOSStrNCopy(license.strKey, strKey, strlen(strKey) + 1, sizeof(license.strKey));
 		if (nRetVal != XN_STATUS_OK)
 		{
-			printf("Error: Key is too long (should be up to %u characters).", sizeof(license.strKey) - 1);
+			printf("Error: Key is too long (should be up to %lu characters).", (unsigned long) (sizeof(license.strKey) - 1));
 			return -1;
 		}
 	}
